components/support/utils/src/main/java/mozilla/components/support/utils/URLStringUtils.kt

```diff
@@ -28,4 +29,12 @@ object URLStringUtils {
         }
         return uri.toString()
     }
+
+    private val isURLLenient by lazy {
+        // Be lenient about what is classified as potentially a URL. Anything that contains a :,
+        // ://, or . and has no internal spaces is potentially a URL. Use java.util.regex because
+        // of its explicit unicode support
+        // (https://docs.oracle.com/javase/7/docs/api/java/util/regex/Pattern.html).
+        Pattern.compile("^\\s*\\w+(://|:|\\.)\\w+\\S*\\s*$")
```

Comment:
Why did we decide to do our own expression over using `java.net.URL` or similar classes? Perhaps we should explain in code comments. It's also possible this is a "back out the regressing change" which is cool by me too. :P

I know `java.net.URL` throws an Exception when a result is not a URL so, due to the performance penalty, it's not appropriate for rapid (negative) matching but I don't know if there are other ways to use it or other possibilities.

Reply:
`java.net.URL` does not recognized as URLs strings that do not begin with "http://" or "https://".